config/condition.go

```diff
@@ -0,0 +1,28 @@
+package config
+
+import (
+	"reflect"
+)
+
+// ConditionConfig configures a condition on a task to define the condition on
+// which to execute a task.
+type ConditionConfig interface {
+	Copy() ConditionConfig
+	Merge(ConditionConfig) ConditionConfig
+	Finalize([]string)
+	Validate() error
+	GoString() string
+}
+
+// DefaultConditionConfig returns the default conditions which is an unconfigured
+// 'services' type condition.
+func DefaultConditionConfig() ConditionConfig {
+	return &ServicesConditionConfig{}
+}
+
+// isNil can be used to check if a ConditionConfig interface is nil by checking
+// both the type and value. Not needed for checking a ConditionConfig
+// implementation i.e. isNil(ConditionConfig), servicesConditionConfig == nil
+func isNil(c ConditionConfig) bool {
+	return c == nil || reflect.ValueOf(c).IsNil()
```

Comment:
No changes requested, just felt like passing along an alternative to IsNil() that can be handy when performance is important (this is about 10x faster). Apologies if already known.

Instead of reflect.ValueOf(c).IsNil() you can use the knowledge of potential types to speed this up with a switch over the types. Something like...

```
switch v := c.(type) {
    case *ServicesConditionConfig:
        result := v == nil
    case  *SomeOtherType:
        result := v == nil
}
```

Reply:
I didn't know this! Thanks for sharing! I think this is a great idea. I pushed a change with your suggestion. I kept the reflect.ValueOf() as the default as a back up... (not expecting to hit that code).

Reply:
Ohh... good idea using the reflect as the backup/default. :+1: